yeat/workflows/snakefiles/Oxford

```diff
@@ -42,57 +26,46 @@ rule nanofilt:
 
 rule nanoplot:
     output:
-        raw_report="seq/nanoplot/{sample}/raw_LengthvsQualityScatterPlot_dot.pdf",
-        filtered_report="seq/nanoplot/{sample}/filtered_LengthvsQualityScatterPlot_dot.pdf"
+        raw_report="seq/nanoplot/{sample}/{readtype}/raw_LengthvsQualityScatterPlot_dot.pdf",
+        filtered_report="seq/nanoplot/{sample}/{readtype}/filtered_LengthvsQualityScatterPlot_dot.pdf"
     input:
-        raw_reads="seq/input/{sample}/combined-reads.fq.gz",
-        filtered_reads="seq/nanofilt/{sample}/highQuality-reads.fq.gz"
+        raw_reads="seq/input/{sample}/{readtype}/combined-reads.fq.gz",
+        filtered_reads="seq/nanofilt/{sample}/{readtype}/highQuality-reads.fq.gz"
     threads: 128
     params:
-        outdir="seq/nanoplot/{sample}"
+        outdir="seq/nanoplot/{sample}/{readtype}"
     shell:
         """
         NanoPlot -t {threads} --fastq {input.raw_reads} -o {params.outdir} --N50 -p raw_ --title "{wildcards.sample} Raw" -f pdf --plots dot
         NanoPlot -t {threads} --fastq {input.filtered_reads} -o {params.outdir} --N50 -p filtered_ --title "{wildcards.sample} Filtered" -f pdf --plots dot
         """
 
 
-def get_readtype_flag(sample):
-    readtype = config["sample_readtype"][sample]
-    if readtype == "nano-raw":
-        return "--nano-raw"
-    elif readtype == "nano-corr":
-        return "--nano-corr"
-    elif readtype == "nano-hq":
-        return "--nano-hq"
-
-
 rule flye:
     output:
-        contigs="analysis/{sample}/{label}/flye/{sample}_contigs.fasta"
+        contigs="analysis/{sample}/{readtype,nano-raw|nano-corr|nano-hq}/{label}/flye/contigs.fasta"
```

Comment:
Wildcard constraints?

Reply:
Yes. Originally, we did not import all rules from all snakefiles into `Workflows.snk`. (We ran each workflow separately--one-at-a-time.) Because there are multiple rules that have the same name and output, I needed to make the wildcard constraints even more stricter by using regex. If the wildcard constrains aren't made, when you import rules that have the same output, snakemake gets confused because it doesn't know which rule to execute.

Here's the documentation for it: https://snakemake.readthedocs.io/en/stable/tutorial/additional_features.html#constraining-wildcards